packages/aws-rfdk/lib/deadline/lib/sep-spotfleet.ts

```diff
@@ -0,0 +1,693 @@
+   * only thing using this limit is the configuration of the listener ports. More than 8 workers can be started,
+   * but only the first 8 will have their ports opened in the workers' security group.
+   */
+  private static readonly MAX_WORKERS_PER_HOST = 8;
+
+  /**
+   * The security groups/rules used to allow network connections.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The principal to grant permissions to.
+   */
+  public readonly grantPrincipal: IPrincipal;
+
+  /**
+   * The stack in which this fleet is defined.
+   */
+  public readonly stack: Stack;
+
+  /**
+   * The environment this resource belongs to.
+   */
+  public readonly env: ResourceEnvironment;
+
+  /**
+   * The port workers listen on to share their logs.
+   */
+  public readonly listeningPorts: Port;
+
+  /**
+   * Security Groups assigned to this fleet.
+   */
+  public readonly securityGroups: ISecurityGroup[];
+
+  /**
+   * The user data that instances use when starting up.
+   */
+  public readonly userData: UserData;
+
+  /**
+   * The operating system of the script host.
+   */
+  public readonly osType: OperatingSystemType;
+
+  /**
+   * An IAM role associated with the instance profile assigned to its resources.
+   */
+  public readonly role: IRole;
+
+  /**
+   * An IAM role that grants the Spot Fleet the permission to request, launch, terminate, and tag instances on your behalf.
+   */
+  public readonly iamFleetRole: IRole;
+
+  /**
+   * Spot Fleet Configurations constructed from the provided input.
+   * Each congiguration is a mapping between one Deadline Group and one Spot Fleet Request Configuration.
+   */
+  public readonly sepSpotFleetRequestConfigurations: any[];
+
+  /**
+   * An id of the worker AMI.
+   */
+  protected readonly imageId: string;
+
+  /**
+   * The tags to apply during creation of instances and of the Spot Fleet Request.
+   */
+  protected readonly tags: TagManager;
+
+  constructor(scope: Construct, id: string, props: SEPSpotFleetProps) {
+    super(scope, id);
+
+    this.stack = Stack.of(scope);
+    this.env = {
+      account: this.stack.account,
+      region: this.stack.region,
+    };
+
+    this.validateProps(props);
+
+    this.securityGroups = props.securityGroups ?? [ new SecurityGroup(this, 'SEPConfigurationSecurityGroup', { vpc: props.vpc }) ];
+    this.connections = new Connections({ securityGroups: this.securityGroups });
+    this.connections.allowToDefaultPort(props.renderQueue.endpoint);
+
+    this.role = props.role ?? new Role(this, 'SpotWorkerRole', {
+      assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('AWSThinkboxDeadlineSpotEventPluginWorkerPolicy'),
+      ],
+      description: `Role for ${id} in region ${this.env.region}`,
+      roleName: 'DeadlineSpot' + this.calculateResourceTagValue([this]),
```

Comment:
We need to be careful here when specifying what CDK refers to as a [physical name](https://docs.aws.amazon.com/cdk/latest/guide/resources.html#resources_physical_names):

> Assigning physical names to resources has some disadvantages in AWS CloudFormation. Most importantly, any changes to deployed resources that require a resource replacement, such as changes to a resource's properties that are immutable after creation, will fail if a resource has a physical name assigned. If you end up in a state like that, the only solution is to delete the AWS CloudFormation stack, then deploy the AWS CDK app again. See the AWS CloudFormation documentation for details.

Is there any reason why we can't just not specify `roleName` and have CloudFormation generate a physical name for us?

Reply:
Requirement of the Spot Event Plugin. It requires that the role name start with "DeadlineSpot"

Reply:
I wonder if we have this requirement only because in `AWSThinkboxDeadlineSpotEventPluginAdminPolicy`
we have:
`            "Resource": [
                "arn:aws:iam::*:role/aws-ec2-spot-fleet-tagging-role",
                "arn:aws:iam::*:role/DeadlineSpot*"
            ]
 ` 
 We might then use the approach we used with the `fleetRole`: just create a new role and add policy to the render queue, but then we would have a few unused statements in the RQ policy, which is not great.
 
 On the other hand, such approach could help customers eliminate the need to create a ResourceTracker role, as we could create a new role from managed policy `AWSThinkboxDeadlineResourceTrackerAccessPolicy` and add policy to the RQ.
 
 What do you think?

Reply:
I wonder if we have this requirement only because in `AWSThinkboxDeadlineSpotEventPluginAdminPolicy`
we have:
`            "Resource": [
                "arn:aws:iam::*:role/aws-ec2-spot-fleet-tagging-role",
                "arn:aws:iam::*:role/DeadlineSpot*"
            ]
 ` 
We might then use the approach we used with the `fleetRole`: just create a new role and add policy to the render queue, but then we would have a few unused statements in the RQ policy, which is not great. I don't think it's worth it.

Reply:
There's no harm in those statements being unused in the managed policy...

Reply:
Seems to be working after allowing `PassRole`. In `AWSThinkboxDeadlineSpotEventPluginAdminPolicy` we also allow `GetRole` for both fleet role and fleet instance role, but it seems that everything is working if I only allow `PassRole`, so I will keep the permissions strict for now.
We will be able to allow GetRole n the future, if something won't work.